### ISSUE TITLE
Images add empty pages

### ISSUE THREAD

**Issue by ehmkah** [asciidoctor/asciidoctorj#457]
We are using asciidoctorj-pdf for exporting our document (https://github.com/ehmkah/cas_swa_sad ) to pdf. Unfortunately we have some problems with the images:

- sometimes the left part is cut off (_05_building_block_view.adoc => 05_bausteinsicht_level1.png => )
- sometimes empty pages are added (_06_runtime_view.adoc  => 06-verkaufsprozess_uebersicht.png)

Any ideas from your experience what we should have in our mind if we use the asciidoctorjpdf?

**Comment by mojavelinux** [comment]
I think this is related to the following issue upstream: https://github.com/asciidoctor/asciidoctor-pdf/issues/388.

What it comes down to is that you are using an inline image when you mean to use a block image. Use two colons after `image` in the image macro instead of one.

Inline images are meant to fit without the line of text. If they exceed that area, then the behavior is undefined.

**Comment by ehmkah** [comment]
For the small images using the block image macro did the trick. Thank you!

The large image I needed to scale down to half the size to prevent asciidoctor-j from adding empty pages. LaTeX managed it to fill the empty space with text which comes behind the image (described Chapter 17 http://mirror.switch.ch/ftp/mirror/tex/info/epslatex/english/epslatex.pdf) For me it seems that asciidoctor does not support something similar to this or does it?

**Comment by mojavelinux** [comment]
> LaTeX managed it to fill the empty space with text which comes behind the image. For me it seems that asciidoctor does not support something similar to this or does it?

No, it does not. Asciidoctor PDF is not intended for that type of advanced formatting. This is due to the fact that the Prawn API is simply not powerful enough to implement that type of behavior. When you can to that level of fine tuning, I recommend either extending the converter to implement custom logic or switch to Asciidoctor LaTeX.

**Comment by mojavelinux** [comment]
> The large image I needed to scale down to half the size to prevent asciidoctor-j from adding empty pages. 

I am noticing cases when Asciidoctor PDF is adding blank pages when trying to place images that are larger than the page. I haven't yet figured out exactly the scenario. If you can give me a very specific example that demonstrates this problem, that would help. Either way, lets follow up in the upstream project => https://github.com/asciidoctor/asciidoctor-pdf.
